libs/wingsdk/src/target-sim/container.inflight.ts

```diff
@@ -194,6 +241,114 @@ export class Container implements IContainerClient, ISimulatorResourceInstance {
     }
   }
 
+  private async docker(
+    command: string,
+    args: string[],
+    options: DockerSpawnOptions = {}
+  ): Promise<string> {
+    const child = this.dockerSpawn(command, args, options);
+    return child.join();
+  }
+
+  private dockerSpawn(
+    command: string,
+    args: string[],
+    options: DockerSpawnOptions = {}
+  ): DockerProcess {
+    let logStdout = options.stdout ?? false;
+    const logStderr = options.stderr ?? false;
+    const onError = options.onError;
+
+    // can be used to hide container logs (used in our end to end tests)
+    if (process.env.WING_HIDE_CONTAINER_LOGS) {
+      logStdout = false;
+    }
+
+    this.addTrace(
+      `Running: docker ${command} ${args.join(" ")}`,
+      TraceType.RESOURCE,
+      LogLevel.VERBOSE
+    );
+
+    const child = spawn("docker", [command, ...args], {
+      cwd: this.props.cwd,
+      stdio: "pipe",
+    });
+
+    let started = true;
+
+    child.once("exit", (code) => {
+      started = false;
+
+      if (onError && code !== 0 && code != null) {
+        onError(new Error(`non-zero exit code: ${code}`));
+      }
+    });
+
+    const output: Buffer[] = [];
+
+    child.stdout.on("data", (data) => {
+      output.push(data);
+
+      if (logStdout) {
+        this.addTrace(data.toString(), TraceType.LOG, LogLevel.INFO);
+      }
+    });
+
+    if (logStderr) {
+      child.stderr.on("data", (data) => {
+        this.addTrace(data.toString(), TraceType.LOG, LogLevel.INFO);
```

Comment:
Should this also have LogLevel.ERROR?

Reply:
I know stderr is also often used for "verbose" output, but I feel like putting it at the same log level as stdout seems off

Reply:
I don't think this should be an ERROR level. Despite it's name "STDERR" is not just for errors. ERROR logs in the Wing Console are going to be displayed with BIG RED HIGHLIGHTS so we should be careful to use them just when there's an actionable error from the system.